More docs onFocus and onBlur events

ctrlaltdylan opened the pull request:
Including more substantial documentation around the `onFocus` and `onBlur` events that aren't so common sense on first look.



<!--

Thank you for the PR! Contributors like you keep React awesome!

Please see the Contribution Guide for guidelines:

https://github.com/reactjs/reactjs.org/blob/master/CONTRIBUTING.md

If your PR references an existing issue, please add the issue number below

-->

iamdarshshah commented on content/docs/reference-events.md:
```diff
@@ -165,6 +165,14 @@ onFocus onBlur
+
+#### onBlur
+
+`onBlur` prop is initiated when focus has left on the element. For example, when the user clicks into and then clicks outside of on a text input element, the function defined with the `onFocus` prop is called.
```
```suggestion
`onBlur` prop is initiated when focus has left on the element. For example, when the user clicks into and then clicks outside of on a text input element, the function defined with the `onBlur` prop is called.
```

gaearon commented:
How do you feel about incorporating this example? Seems like it’s a useful collection of behaviors people want. https://github.com/facebook/react/issues/6410#issuecomment-671915381

eps1lon commented on content/docs/reference-events.md:
```diff
@@ -165,6 +165,14 @@ onFocus onBlur
 
+#### onFocus
+
+The `onFocus` prop is initiated when focus is on the element. For example, when the user clicks on a text input element, the function defined with the `onFocus` prop is called.
```
That is certainly true for chrome on windows and linux as far as I know. But does not apply to e.g. firefox macOS. 

More specifically: By spec it is up to the user agent (e.g. a browser) to determine whether a clicked element is also focused.

eps1lon replied:
Or it might be true for inputs. I was a bit suspicious because of the "click-focus" association which is a bit tricky on other platforms.

ctrlaltdylan replied:
@eps1lon I just booted up Firefox + Mac and ran through a sample:

https://www.loom.com/share/34f1a18300b44187b7a65e2c01b73f8b

Sandbox: https://codesandbox.io/s/busy-browser-0me16?file=/src/App.js:146-191

It _appears_ to be 1:1 to Chrome, at least for the basic examples I wanted to document. If you find something else different please feel free to edit/tack on.

ctrlaltdylan commented:
> How do you feel about incorporating this example? Seems like it’s a useful collection of behaviors people want. [facebook/react#6410 (comment)](https://github.com/facebook/react/issues/6410#issuecomment-671915381)

@gaearon  Forked your explanation and changed the logging events to be a tad more people friendly and specific. Added a few more examples to cover vanilla `onFocus` and `onBlur` on a solo element.

iamdarshshah commented on content/docs/reference-events.md:
```diff
@@ -165,11 +165,82 @@ onFocus onBlur
+      }}
+      onBlur={(e) => {
+        console.log("left focus of", e.target);
+        if (e.currentTarget === e.target) {
```
I think this condition needs to be changed, because `e.currentTarget === e.target` will never be true for `onBlur` events as it is triggered when _self_ loses its focus or when the user leaves a specific field. Thus, this won't be true in any case.  

We can update it to this: 
```suggestion
        if (e.currentTarget !== e.target) {
```
> This will be triggered when we not focus on _self_ element anymore.

gaearon replied:
I'm not sure what you mean but this condition is definitely true when you're blurring the node itself. https://codesandbox.io/s/strange-albattani-7tqr7?file=/src/App.js The proposed edit is wrong and contradicts what the code is supposed to do.

iamdarshshah reviewed (approved):
LGTM! 👍

gaearon commented:
Code blocks need to be consistently indented and should use JS language

ctrlaltdylan commented:
@gaearon thanks, missed that. Updated `onBlur` and `onFocus` examples with the `javascript` tag and fixed spacing so it matches the other examples.
